sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -278,7 +279,13 @@ def _read_side_inputs(self, tags_and_types):
         if not isinstance(si, operation_specs.WorkerSideInputSource):
           raise NotImplementedError('Unknown side input type: %r' % si)
         sources.append(si.source)
-      iterator_fn = sideinputs.get_iterator_fn_for_sources(sources)
+
+      si_counter = opcounters.SideInputReadCounter(
+          self.counter_factory, self.state_sampler,
+          # Inputs are 1-indexed, so we add 1 to i in the side input id
+          counters.side_input_id(self.operation_name, i+1))
+      iterator_fn = sideinputs.get_iterator_fn_for_sources(
+          sources, read_counter=si_counter)
```

Comment:
<!--new_thread; commit:d2ca52a83815c31b329d1e1409344f81482d45c5; resolved:0-->
Do we want to use the same counter for all the sources here ?

Reply:
Yes. A single side input may have different sources, but we want to track bytes/msecs for the side input, not per-source.